File: case-management/internal/server/validate.go
package server

import (
	"errors"
	"slices"
)

func validateSeverity(s string) error {
	if slices.Contains([]string{"low", "medium", "high", "critical"}, s) {
		return nil
	}
	return errors.New("invalid severity")
}

func validateStatus(s string) error {
	if slices.Contains([]string{"new", "triaged", "investigating", "contained", "resolved", "closed"}, s) {
		return nil
	}
	return errors.New("invalid status")
}

func validateResolution(s string) error {
	if slices.Contains([]string{"true_positive", "false_positive", "benign", "informational", "other"}, s) {
		return nil
	}
	return errors.New("invalid resolution")
}

File: case-management-rs/src/validate.rs
const VALID_SEVERITIES: &[&str] = &["low", "medium", "high", "critical"];
const VALID_STATUSES: &[&str] = &["new", "triaged", "investigating", "contained", "resolved", "closed"];
const VALID_RESOLUTIONS: &[&str] = &["true_positive", "false_positive", "benign", "informational", "other"];

pub fn validate_severity(s: &str) -> Result<(), &'static str> {
    if VALID_SEVERITIES.contains(&s) {
        Ok(())
    } else {
        Err("invalid severity")
    }
}

pub fn validate_status(s: &str) -> Result<(), &'static str> {
    if VALID_STATUSES.contains(&s) {
        Ok(())
    } else {
        Err("invalid status")
    }
}

pub fn validate_resolution(s: &str) -> Result<(), &'static str> {
    if VALID_RESOLUTIONS.contains(&s) {
        Ok(())
    } else {
        Err("invalid resolution")
    }
}
